contracts/KeysManager.sol

```diff
@@ -461,6 +468,7 @@ contract KeysManager is EternalStorage, IKeysManager {
         _setMiningKeyByVoting(votingKey, _key);
         _setMiningKeyByPayout(payoutKey, _key);
         _getValidatorMetadata().moveMetadata(_oldMiningKey, _key);
+        _setHasMiningKeyBeenRemoved(_key);
```

Comment:
This must be `_setHasMiningKeyBeenRemoved(_oldMiningKey);` because we swap the `_oldMiningKey` to the `_key`.

Reply:
Yes, you are right, it's my fault